freqtrade/strategy/interface.py

```diff
@@ -707,35 +719,42 @@ def _analyze_ticker_internal(self, dataframe: DataFrame, metadata: dict) -> Data
         # always run if process_only_new_candles is set to false
         if (not self.process_only_new_candles or
                 self._last_candle_seen_per_pair.get(pair, None) != dataframe.iloc[-1]['date']):
+
+            populate_indicators = not external_data
```

Comment:
I think we want to allow populating of indicators within the strategy even if it is fetching data externally, so the strategy can pull in that dataframe and determine what to do with it. The the emitting/broadcasting would only be done if it's configured to do so, instead of coupling the population of indicators with emitting/broadcasting as a leader. I think that would largely remove the need to pass `external_data` through a lot of these functions.

Reply:
Could we pull it from the config in the strategy?
```
self.config.get('api_server', {}).get('enable_message_ws', False):
```

Reply:
See this comment [https://github.com/freqtrade/freqtrade/pull/7303#discussion_r959843287](https://github.com/freqtrade/freqtrade/pull/7303#discussion_r959843287)

Reply:
why would you disable broadcasting? the websocket itself will always be up (i intend to use that for getting uptodate info to the ui) - so you're really not saving much.

Reply:
Broadcasting the dataframe is only disabled if we are a follower. Anything and everything else will still work. The only reason we don't broadcast dataframes as a follower is unnecessary overhead of serializing the dataframe for no reason.